src/libraries/System.Net.Http/src/System/Net/Http/Headers/KnownHeaders.cs

```diff
@@ -40,7 +40,7 @@ internal static class KnownHeaders
         public static readonly KnownHeader ContentRange = new KnownHeader("Content-Range", HttpHeaderType.Content | HttpHeaderType.NonTrailing, GenericHeaderParser.ContentRangeParser, null, H2StaticTable.ContentRange);
         public static readonly KnownHeader ContentSecurityPolicy = new KnownHeader("Content-Security-Policy", http3StaticTableIndex: H3StaticTable.ContentSecurityPolicyAllNone);
         public static readonly KnownHeader ContentType = new KnownHeader("Content-Type", HttpHeaderType.Content | HttpHeaderType.NonTrailing, MediaTypeHeaderParser.SingleValueParser, null, H2StaticTable.ContentType, H3StaticTable.ContentTypeApplicationDnsMessage);
-        public static readonly KnownHeader Cookie = new KnownHeader("Cookie", H2StaticTable.Cookie, H3StaticTable.Cookie);
+        public static readonly KnownHeader Cookie = new KnownHeader("Cookie", HttpHeaderType.Request, CookieHeaderParser.Parser, null, H2StaticTable.Cookie, H3StaticTable.Cookie);
```

Comment:
Note to reviewers: Adding `HttpHeaderType.Request` is a breaking change as you can no longer add this header to the content/response collections with validating APIs.

We may still want to do it, just making sure it's a deliberate decision.

Reply:
@MihaZupan I changed the header type to `HttpHeaderType.Custom` for now. Thanks